File: apps/messenger/src/modules/discord/models/discord-message.dto.ts
import { DiscordMessageDTOOptions } from "../types/discord-message-dto-options.type";
import { DiscordMessageAuthorDTO } from "./discord-message-author.dto";

export class DiscordMessageDTO {
  public readonly applicationId?: string;
  public readonly author: DiscordMessageAuthorDTO;
  public readonly cleanContent: string;
  public readonly content: string;
  public readonly createdAt: Date;
  public readonly hasThread: boolean;
  public readonly id: string;
  public readonly pinnable: boolean;
  public readonly pinned: boolean;
  public readonly position?: number;
  public readonly system: boolean;
  public readonly url: string;
  public readonly guildId: string;
  public readonly channelId: string;
  public readonly deleted: boolean;

  public constructor(options: DiscordMessageDTOOptions) {
    this.applicationId = options.applicationId;
    this.author = options.author;
    this.cleanContent = options.cleanContent;
    this.content = options.content;
    this.createdAt = options.createdAt;
    this.hasThread = options.hasThread;
    this.id = options.id;
    this.pinnable = options.pinnable;
    this.pinned = options.pinned;
    this.position = options.position;
    this.system = options.system;
    this.url = options.url;
    this.guildId = options.guildId;
    this.channelId = options.channelId;
    this.deleted = options.deleted;
  }
}

File: apps/builder/src/main/kotlin/dev/felipeflohr/w2gservices/builder/dto/DiscordMessageDTO.kt
package dev.felipeflohr.w2gservices.builder.dto

import dev.felipeflohr.w2gservices.builder.annotations.NoArg
import dev.felipeflohr.w2gservices.builder.entities.DiscordMessageEntity
import java.util.Date

@NoArg
data class DiscordMessageDTO(
    var applicationId: String?,
    var author: DiscordMessageAuthorDTO,
    var cleanContent: String,
    var content: String,
    var createdAt: Date,
    var hasThread: Boolean,
    var id: String,
    var pinnable: Boolean,
    var pinned: Boolean,
    var position: Int?,
    var system: Boolean,
    var url: String,
    var guildId: String,
    var channelId: String,
    var deleted: Boolean,
) {
    fun toEntity(): DiscordMessageEntity {
        return DiscordMessageEntity(
            messageCreatedAt = createdAt,
            messageId = id,
            url = url,
            pinned = pinned,
            system = system,
            content = content,
            guildId = guildId,
            position = position,
            channelId = channelId,
            author = author.toEntity(),
            id = null,
            delimitationMessage = null,
            fileLogs = null,
            fileReferences = null,
        )
    }
}
